include/tunnler/room.h

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 Tunnler Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+//
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "tunnler/tunnler.h"
+
+#include "RakPeerInterface.h"
+
+const uint16_t DefaultRoomPort = 1234;
+
+// This is what a server [person creating a server] would use.
+class Room final {
+public:
+    enum class State {
+        Open,   // The room is open and ready to accept connections.
+        Closed, // The room is not opened and can not accept connections.
+    };
+
+    struct Member {
+        std::string nickname; ///< The nickname of the member.
+        MacAddress mac_address; ///< The assigned mac address of the member.
+        RakNet::AddressOrGUID network_address; ///< The network address of the remote peer.
+    };
+
+    using MemberList = std::vector<Member>;
```

Comment:
We now have Room::MemberList and RoomMember::MemberList, both holding almost the same information. I know technically there is no problem with that, but going to the code sometimes it's hard to distinguish which one is meant. Maybe rename one to ServerMemberList or similar or use a std::map<RakNet::AddressOrGUID, RoomMember::MemberInformation> since we have to store the complete MemberInformation on the Server anyway.